theano/sandbox/cuda/basic_ops.py

```diff
@@ -2614,6 +2618,52 @@ def perform(self, node, inp, out_):
                 o[j] = x[i]
             out[0] = o
 
+    def c_code(self, node, name, inputs, outputs, sub):
+        x, idx = inputs
+        out, = outputs
+        fail = sub['fail']
+        if node.inputs[0].ndim not in [2, 3]:
+            raise NotImplementedError("This case does not have C code yet.")
+        if node.inputs[1].dtype != 'int64':
+            raise Exception("Index should have dtype int64. Check this node make_node().")
+        return """
+        int max_threads=512;
+        //take(idx, 0, out, "raise", max_threads);
+        PyObject * out_ = NULL;
+        PyObject * ret = NULL;
+        Py_INCREF(%(x)s);
+        PyObject * args = PyTuple_New(5);
+        PyObject * zero = PyInt_FromLong(0);
+        PyObject * max = PyInt_FromLong(max_threads);
+        PyObject * raise = PyString_FromString("raise");
+        if(args == NULL || zero == NULL || max == NULL || raise == NULL){
+            %(fail)s;
+        }
+
+        out_ = (PyObject *) %(out)s;
+        if (out_ == NULL)
+            out_ = Py_None;
+        else
+            Py_INCREF(out_);
+        PyTuple_SetItem(args, 0, (PyObject *) %(idx)s);
+        PyTuple_SetItem(args, 1, zero);
+        PyTuple_SetItem(args, 2, out_);
+        PyTuple_SetItem(args, 3, raise);
+        PyTuple_SetItem(args, 4, max);
```

Comment:
There is already tests for that inside sandbox/cuda/test_basic_ops.py:T_subtensor

I added check and fixed a segfault (I forgot to incref idx before setitem)